GPII-2912: Updated to work with upcoming "potentia ii" version of Infusion.

the-t-in-rtf opened the pull request:
See [GPII-2912](https://issues.gpii.net/browse/GPII-2912) for details.

the-t-in-rtf committed:
GPII-2912: Updated to work with upcoming "potentia ii" version of Infusion.

the-t-in-rtf committed:
GPII-2912: Updated coverage client to work better with upcoming "potentia ii" version of Infusion.

the-t-in-rtf committed:
GPII-2912: Updated dependencies.

the-t-in-rtf committed:
GPII-2993: Fixed broken log level.

the-t-in-rtf committed:
Committed updated package-lock.json

the-t-in-rtf commented:
This pull now also includes the fix for the "fluid.LogLevel" typo that likely resulted in some of the `undefined` chatter we discussed [in the gpii-app pull](https://github.com/GPII/gpii-app/pull/44).

the-t-in-rtf committed:
GPII-2994: Added ability to configure rimraf using declarative options.
NOJIRA: Updated to latest fluid-config-eslint and fixed linting errors.

the-t-in-rtf commented on src/js/client/coverageSender.js:
```diff
@@ -15,85 +15,55 @@
+(function (Testem) {
     "use strict";
+    // Pure JS equivalent of several fluid.registerNamespace calls.
+    window.gpii = { testem: { coverage: {} } };
```
I discovered a deep flaw with this today that would have bitten us soon enough.  It's only truly safe if you're working outside the gpii namespace or loading the coverage client absolutely last.  Otherwise it's very easy for it to delete existing namespaced material.  I'll clean this up before I cut the release.

the-t-in-rtf committed:
GPII-2912: Fixed "namespace bomb" in coverage sender.

the-t-in-rtf committed:
GPII-2912: Updated to potentia-ii version of gpii-express.

amb26 commented on src/js/client/coverageSender.js:
```diff
@@ -15,85 +15,57 @@
+    // Pure JS equivalent of a fluid.registerNamespace call.
+    window.gpii = window.gpii || {};
+    window.gpii.testem = window.gpii.testem || {};
+    window.gpii.testem.coverage = window.gpii.testem.coverage || {};
```
"It's how we live"

amb26 commented:
It would be helpful if the coverage reporter was updated to produce html reports as it does with our other projects. If it was, it would help in spotting interesting issues such as that gpii.testem.instrumenter.resolveRelativePattern in instrumenter.js is never called. Also it seems that most of gpii.testem.instrumenter.runner.instrumentThenExit is not either.

the-t-in-rtf commented:
@amb26, the flags passed to nyc should also have resulted in an "lcov" report which I assumed would include the HTML we're used to.  I don't see any evidence of that, though.  I'll create an .nycrc config file based on a known working example and confirm that a report is generated.  I'll also prune unused code or add tests depending.

the-t-in-rtf committed:
GPII-2912: Fixed NYC configuration and removed unreachable code leftover from the gpii-launcher runner.

the-t-in-rtf commented:
@amb26, I fixed the reporting configuration and removed the unused code and component defaults. 
 All were leftover from the defunct gpii-launcher-based "instrumentation runner".  You can see an example of the updated report here:

http://the-t-in-rtf.github.io/coverage-reports/gpii-testem/20180605/

Nearly all of the branch and line coverage gaps are untested failure modes now.
